src/Engine/ProtoImperative/CodeGen.cs

```diff
@@ -91,6 +91,41 @@ private ProtoCore.DSASM.CodeBlock GetDeltaCompileCodeBlock()
             return cb;
         }
 
+        /// <summary>
+        /// Emits a block of code for the following cases:
+        ///     if elseif else body
+        ///     while body
+        ///     for body
+        /// </summary>
+        /// <param name="codeBlock"></param>
+        /// <param name="inferedType"></param>
+        /// <param name="subPass"></param>
+        private void EmitCodeBlock(
+            List<ImperativeNode> codeBlock, 
+            ref ProtoCore.Type inferedType, 
+            bool isBooleanOp = false, 
```

Comment:
Sure, although currently there are hundreds of lines of code still referring to this. Will assign this task to an intern: http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7452